rfc/0054-sdk3-management-apis.md

```diff
@@ -3569,6 +3611,18 @@ enum StorageBackend {
 
 * `StorageBackend` ([`StorageBackend`](#storagebackend)) - The storage type to use (note: Magma is EE only).
 
+* `HistoryRetentionCollectionDefault` (`bool`) - Whether to enable history retention on collections by default.
+  * URL query field: `historyRetentionCollectionDefault`
+  * If the SDK does not have the ability to differentiate between not set and `false` then a different type should be used instead, suggested:
+    * `enum HistoryRetentionCollectionDefaultSettings { ON, OFF }`
+
+* `HistoryRetentionBytes` (`int`) - The maximum size, in bytes, of the change history that is written to disk for all collections in this bucket.
+  * URL query field: `historyRetentionBytes`
+
+* `HistoryRetentionDuration` (`duration`) - The maximum number of seconds to be covered by the change history that is written to disk for all collections in this bucket.
+  * URL query field: `historyRetentionSeconds` sent as an int seconds value
```

Comment:
Do we want to group these 3 fields behind a HistoryRetention object containing bytes and duration?

Reply:
I was avoiding doing that in Go as I've already had to do some hackery for HistoryRetentionCollectionDefault due to false and not set being the same value so adding another object on top would be quite verbose.

Reply:
Does `HistoryRentionBytes` and/or `HistoryRentionDuration` need to be set if `HistoryRententionCollectionDefault` is `true`?  Or do we have default values that should be used?

Reply:
The docs state that if neither are set then History will be ignored. I've tested and server does not error if HistoryRententionCollectionDefault is true and both bytes and duration are empty. I'd say we leave it to the server.